src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -38,17 +38,168 @@ namespace System.Text.Json
 
             ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
 
-            if (_stringHasEscaping)
+            if (ValueIsEscaped)
             {
-                int idx = span.IndexOf(JsonConstants.BackSlash);
-                Debug.Assert(idx != -1);
-                return JsonReaderHelper.GetUnescapedString(span, idx);
+                return JsonReaderHelper.GetUnescapedString(span);
             }
 
             Debug.Assert(span.IndexOf(JsonConstants.BackSlash) == -1);
             return JsonReaderHelper.TranscodeHelper(span);
         }
 
+        /// <summary>
+        /// Copies the current JSON token value from the source, unescaped as a UTF-8 string to the destination buffer.
+        /// </summary>
+        /// <param name="utf8Destination">A buffer to write the unescaped UTF-8 bytes into.</param>
+        /// <returns>The number of bytes written to <paramref name="utf8Destination"/>.</returns>
+        /// <remarks>Unlike <see cref="GetString"/>, this method does not support <see cref="JsonTokenType.Null"/>.</remarks>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if trying to get the value of the JSON token that is not a string
+        /// (i.e. other than <see cref="JsonTokenType.String"/> or <see cref="JsonTokenType.PropertyName"/>.
+        /// <seealso cref="TokenType" />
+        /// It will also throw when the JSON string contains invalid UTF-8 bytes, or invalid UTF-16 surrogates.
+        /// </exception>
+        /// <exception cref="ArgumentException">Thrown if the destination buffer is too small to hold the unescaped value.</exception>
```

Comment:
```suggestion
        /// <exception cref="ArgumentException">The destination buffer is too small to hold the unescaped value.</exception>
```

According to https://github.com/dotnet/dotnet-api-docs/wiki/Exceptions#how-to the text here starts after an implicit "T is thrown if", so "thrown if" shouldn't be in the message.